sakiparens.py

```diff
@@ -0,0 +1,17 @@
+from __future__ import unicode_literals
+
+def parens(string):
+    stack = []
+    for char in string:
+        if char == '(':
+            stack.append(char)
+        elif char == ')':
+            try:
+                stack.pop()
```

Comment:
@cewing I thought popping from the end of a list ran in constant time, and that pop(0) was horrendous?